### PR TITLE
figma: init at 0.8.1

### PR DESCRIPTION
**Pull request by andyrichardson**
###### Motivation for this change

Add [Figma](https://www.figma.com/) built from [figma-linux](https://github.com/Figma-Linux/figma-linux) client.

There are features not available in the browser version of Figma such as custom font support hence the need for a native client.

###### Notes

 - Originally was building this from source - can provide that version if that could help add support for more platforms+architectures
 -  Whether building from source or with prebuilt binaries, I wasn't able to get this working without using FHS [(forum discussion here)](https://discourse.nixos.org/t/could-use-some-help-packaging-a-binary/14801). **Would appreciate any help.**
 - See  code comments regarding licensing

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/figma/default.nix [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    mkdir -p $out/bin
+    cp -r ${figma-fhs}/bin/figma-fhs $out/bin/figma
+
+    # Link icons
+    mkdir -p $out/share/icons
+    cp -r ${figma-exec}/share/icons/. $out/share/icons/
+
+    # Link desktop item
+    mkdir -p $out/share/applications
+    ln -s ${desktop-item}/share/applications/* $out/share/applications
```
Comment (legendofmiracles) [inline pkgs/applications/graphics/figma/default.nix:141]: use the desktopItem attribute instead

Comment (andyrichardson) [reply pkgs/applications/graphics/figma/default.nix]: @legend thanks for the suggestion - amended :+1:

File: pkgs/applications/graphics/figma/default.nix [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      xorg.libXrandr
+      xorg.libxshmfence
+      wayland
+    ];
+    runScript = "figma";
+  };
+
+in stdenv.mkDerivation {
+  inherit version;
+  name = "figma";
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/figma/default.nix:114]:
```suggestion
  pname = "figma";
  inherit version;
```

File: pkgs/applications/graphics/figma/default.nix [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+let
+  version = "0.8.1";
+  # Figma executable.
+  # Currently won't run outside of FHS even with autopatching - needs help.
+  figma-exec = stdenv.mkDerivation rec {
+    inherit version;
+    pname = "figma-exec";
+    src = pkgs.fetchurl {
+      url =
+        "https://github.com/Figma-Linux/figma-linux/releases/download/v${version}/figma-linux_${version}_linux_amd64.zip";
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/figma/default.nix:16]:
```suggestion
    src = fetchurl {
      url = "https://github.com/Figma-Linux/figma-linux/releases/download/v${version}/figma-linux_${version}_linux_amd64.zip";
```

File: pkgs/applications/graphics/figma/default.nix [hunk 4]
```diff
@@ -0,0 +1,133 @@
+{ pkgs, lib
+# Specify any font packages to include
+# e.g. figma.override { fonts = [ noto-fonts fira-code ]; }
+, fonts ? [ ], ... }:
+
+with pkgs;
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/figma/default.nix:6]:
```suggestion
```
That shouldn't be done.

File: pkgs/applications/graphics/figma/default.nix [hunk 5]
```diff
@@ -0,0 +1,133 @@
+{ pkgs, lib
+# Specify any font packages to include
+# e.g. figma.override { fonts = [ noto-fonts fira-code ]; }
+, fonts ? [ ], ... }:
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/figma/default.nix:4]:
```suggestion
, fonts ? [ ]
}:
```

File: pkgs/applications/graphics/figma/default.nix [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+      ${lib.concatMapStringsSep "\n"
+      (f: "cp -r ${f}/share/fonts/. $out/share/fonts/") fonts}
+
+      # Link binary
+      mkdir -p $out/bin
+      ln -s $out/etc/figma-linux/figma-linux $out/bin/figma
+
+      # Link desktop item
+      mkdir -p $out/share/applications
+      ln -s ${desktopItem}/share/applications/* $out/share/applications
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/figma/default.nix:50]: Please use copyDesktopItems instead.

**Comment by 18kimn** [comment]
hello! are there perhaps any updates on this PR? Would love to see figma on nixos. I'm a new user but can assist if given direction

**Comment by andyrichardson** [comment]
Closing as I no longer use nix. If others want to contribute, this can still be used as a starting point 👍

**Pull request closed without merging**